cmc-ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/PartyMapper.java

```diff
@@ -0,0 +1,116 @@
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCompany;
+import uk.gov.hmcts.cmc.ccd.domain.CCDIndividual;
+import uk.gov.hmcts.cmc.ccd.domain.CCDOrganisation;
+import uk.gov.hmcts.cmc.ccd.domain.CCDParty;
+import uk.gov.hmcts.cmc.ccd.domain.CCDPartyType;
+import uk.gov.hmcts.cmc.ccd.domain.CCDSoleTrader;
+import uk.gov.hmcts.cmc.ccd.exception.MappingException;
+import uk.gov.hmcts.cmc.domain.models.party.Company;
+import uk.gov.hmcts.cmc.domain.models.party.Individual;
+import uk.gov.hmcts.cmc.domain.models.party.Organisation;
+import uk.gov.hmcts.cmc.domain.models.party.Party;
+import uk.gov.hmcts.cmc.domain.models.party.SoleTrader;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+@Component
+public class PartyMapper implements Mapper<CCDParty, Party> {
+
+    private final IndividualMapper individualMapper;
+    private final CompanyMapper companyMapper;
+    private final OrganisationMapper organisationMapper;
+    private final SoleTraderMapper soleTraderMapper;
+    private final AddressMapper addressMapper;
+    private final RepresentativeMapper representativeMapper;
+
+    @Autowired
+    public PartyMapper(IndividualMapper individualMapper, CompanyMapper companyMapper,
+                       OrganisationMapper organisationMapper, SoleTraderMapper soleTraderMapper,
+                       AddressMapper addressMapper, RepresentativeMapper representativeMapper) {
+        this.individualMapper = individualMapper;
+        this.companyMapper = companyMapper;
+        this.organisationMapper = organisationMapper;
+        this.soleTraderMapper = soleTraderMapper;
+        this.addressMapper = addressMapper;
+        this.representativeMapper = representativeMapper;
+    }
+
+    @Override
+    public CCDParty to(Party party) {
+        final CCDParty.CCDPartyBuilder builder = CCDParty.builder();
+
+        if (party instanceof Individual) {
+            builder.type(CCDPartyType.INDIVIDUAL);
+            Individual individual = (Individual) party;
+            builder.individual(individualMapper.to(individual));
+        } else if (party instanceof Company) {
+            builder.type(CCDPartyType.COMPANY);
+            Company company = (Company) party;
+            builder.company(companyMapper.to(company));
+        } else if (party instanceof Organisation) {
+            builder.type(CCDPartyType.ORGANISATION);
+            Organisation organisation = (Organisation) party;
+            builder.organisation(organisationMapper.to(organisation));
+        } else if (party instanceof SoleTrader) {
+            builder.type(CCDPartyType.SOLE_TRADER);
+            SoleTrader soleTrader = (SoleTrader) party;
+            builder.soleTrader(soleTraderMapper.to(soleTrader));
+        }
+
+        return builder.build();
+    }
+
+    @Override
+    public Party from(CCDParty ccdParty) {
+        switch (ccdParty.getType()) {
+            case COMPANY:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1151)